### ISSUE TITLE
unable to unload a filament whilst printing

### ISSUE THREAD

**Issue by robinbon** [bambulab/BambuStudio#4094]
### Bambu Studio Version

1.9.1.67

### Where is the application from?

Bambu Lab Official website

### OS version

mac os

### Additional system information

m1 macbook pro, 16gb ram, 1tb hd

### Printer

x1c with ams

### How to reproduce

Unloading a filament whilst printing job.
In past I have been able to pause a print, unload a filament from AMS and then load a new reel in place.
X1C
AMS 
The ability to unload is greyed out.

### Actual results

greyed out load option whilst printing (paused)

### Expected results

should be able to unload filament

### Project file & Debug log uploads

all projects are effected.
[smart car wheel nut caps v2.3mf.zip](https://github.com/bambulab/BambuStudio/files/15344010/smart.car.wheel.nut.caps.v2.3mf.zip)

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by walterwongbbl** [comment]
@robinbon 
Hi. You should specify the desired filaments from the beginning. We currently align the behavior of studio with that of the printer, so we do not support doing so at this time.

**Comment by robinbon** [comment]
No filament is unable to uload from AMS whist paused printing

Option is greyed out on LCD.

Friend has same problem with their printer, though they tried another slicer ORCA and that worked as it should.

So back to you, unable to unload filament whilst printing.

**Comment by mano-id** [comment]
Same here. Worked before, now It doesn't.
Got a tangled filament. Paused the print. But can't unload. Have to stop the whole print to do It.
See enclosed photo.
![IMG_0091](https://github.com/bambulab/BambuStudio/assets/42246417/7d9221aa-1620-48fa-820a-d3add7d7aa75)

**Comment by robinbon** [comment]
thanks for photo, exactly problem i have.

**Comment by caprageon** [comment]
Hi !

Same here with same config (X1C with AMS).
Load/Unload buttons are greyed out in BambuLab and on screen when the print is paused.

X1C fw : 01.08.00.00
AMS fw: 00.00.06.44
Bambu Studio : 1.9.3.50

First time I tried this so can't confirm it worked in the past.

**Comment by robinbon** [comment]
> Hi !
> 
> Same here with same config (X1C with AMS). Load/Unload buttons are greyed out in BambuStudio and on screen when the print is paused.
> 
> X1C fw : 01.08.00.00 AMS fw: 00.00.06.44 Bambu Studio : 1.9.3.50
> 
> First time I tried this so can't confirm it worked in the past.

it did but but that was a while ago.

**Comment by caprageon** [comment]
I suspect it is intended behavior on the printer side (which I agree is disappointing)
Pretty sure things won't change in BambuStudio unless it is added (back) in printer fw...

**Comment by robinbon** [comment]
> I suspect it is intended behavior on the printer side (which I agree is disappointing) Pretty sure things won't change in BambuStudio unless it is added (back) in printer fw...

No, it used to work, Orca slicer does not have this problem. You should be able to unload a filament whilst printing.

**Comment by walterwongbbl** [comment]
Please use the latest version

**Comment by robinbon** [comment]
yes and this does not work

**Comment by feyzo** [comment]
I have the exact problem with the latest firmware.

I forgot to change filaments and realized my mistake 60% in a 13 hour print. I was able to save the it by cutting the currently loaded filament, letting printer detect the broken filament and replaced with the correct one. Not ideal but I caught it just before the visible parts were being added with the second color.

**Comment by robinbon** [comment]
thanks for comment, shame this option has been broken for so long now.

**Comment by nbenns** [comment]
Verified still an issue with:
Bambu Studio: 1.10.2
X1C FW: 1.08.02.00
AMS FW: 00.00.06.44

I got lucky once and was able to swap colors during a pause because a color I didn't need to change was the last used.
Then I got unlucky the second time.
I didn't realize it was an issue until I found this post, I had never thought this would ever be a problem.

**Comment by nbenns** [comment]
@walterwongbbl do you know if there is a G-Code work around for this issue?
Something to insert the Unload, Pause and then Reload?

**Comment by robinbon** [comment]
@walterwongbbl 
 I dont understand why this bug has been left for so long, it's almost a year since I reported it and it's been ignored.
Orca slicer it was fixed a long time ago.
